numpy/core/src/multiarray/ctors.c

```diff
@@ -3793,7 +3786,6 @@ PyArray_FromFile(FILE *fp, PyArray_Descr *dtype, npy_intp num, char *sep)
                 (skip_separator) fromfile_skip_separator, NULL);
     }
     if (ret == NULL) {
-        Py_DECREF(dtype);
```

Comment:
Just to be clear, this line is likely the only actual bug. Everything else is no just cleanup...